stats/worker.go

```diff
@@ -313,3 +315,37 @@ func isCumulative(v *View) bool {
 	}
 	return false
 }
+
+func deepCopy(rows []*Row) []*Row {
+	// TODO(jbd): Find a better solution so each time
+	// we add a new aggreagtion data type, we don't have
+	// to modify this function.
+	newRows := make([]*Row, 0, len(rows))
+	for _, r := range rows {
+		var cloned AggregationData
+		switch d := r.Data.(type) {
+		case *CountData:
+			cloned = &(*d)
+		case *DistributionData:
+			counts := make([]int64, len(d.CountPerBucket))
+			copy(counts, d.CountPerBucket)
+			c := *d
+			c.CountPerBucket = counts
+			cloned = &c
+		case *SumData:
+			cloned = &(*d)
+		case *MeanData:
+			cloned = &(*d)
+		default:
+			// Panic, so we can easily tell we forgot
+			// to add support for the aggregation data type
+			// during development time.
+			panic("deep copy failed for unknown type")
```

Comment:
Will this break user's binaries when we add a new aggregation type? Shall we log this instead?